shopsafe-backend/src/main/java/com/google/sps/data/StoreDatastoreHandler.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory.Builder;
+
+/*
+ * Class that modifies Store entries in datastore.
+ */
+public class StoreDatastoreHandler {
+    
+    private String id;
+    private Key key;
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    public StoreDatastoreHandler(String id) {
+        this.id = id;
+        this.key = new Builder("Store", id).getKey();
```

Comment:
Does this get the key, and create one, if this is a new id?

Reply:
yeah it creates a new key each time, but that doesnt mean it creates a new datastore entry, its just a way to communicate with datastore